PCB-testing/chris-js/pca-test.js

```diff
@@ -12,31 +12,30 @@
  *
  */
 
-// Get dependencies, initialize stuff
-var sleep = require('sleep');
-/*
-	// If the defaults ever need to get changed, uncomment this
-	// and comment out 'var makePwm = require('adafruit-pca9685');'
-	// Note that the contents of this object are the defaults
+// Should cycle a thruster on and off.
 
-var makePwm =  {
-    "freq": "50",				// frequency of the device
-    "correctionFactor": "1.0",	// correction factor - fine tune the frequency
-    "address": "0x40",			// i2c bus address
-    "device": '/dev/i2c-1',		// device name
-    "debug": <null>				// adds some debugging methods if set
-}
+var i2cBus = require("i2c-bus");
+var Pca9685Driver = require("pca9685").Pca9685Driver;
 
-*/
-var makePwm = require('adafruit-pca9685');
-var pwm = makePwm();
+var options = {
+     i2c: i2cBus.openSync(1),
+     address: 0x40,
+     frequency: 50,
+     debug: false
+};
 
-// Tell PCA to do the things, print to console
+// Loops through turning on and off
 while(true) {
-	console.log('Setting PWM on channel 1 to 255!');
-	pwm.setPulse(1, 255);
-	sleep.sleep(5);
-	console.log('Setting PWM on channel 1 to 0!');
-	pwm.setPulse(1, 0);
-	sleep.sleep(5);
+    var on = setInterval(on(), 1000);
+    var off = setInterval(off(), 1000);
```

Comment:
This will call both of these functions at the same time every 1000ms. try setting the intervals in a promise chain or using `async`. You can put a `setTimeout` on a promise's `resolve` function to make it wait however long you need.

ALSO `setInterval` takes a reference to a function as its first argument. What's written here will actually call `on` and `off` and then set an interval on whatever they return. You can pass them by reference by just getting rid of the parentheses.

ALSO ALSO be careful about variable naming. Functions count as variables too because JS is made of goopy jello. Here I think node interprets the `on` and `off` functions first and then sees the variables and is like OMFG WAT `on` AND `off` ALREADY EXiST